### PR TITLE
Slight on_move.dm rewrite

### PR DESCRIPTION
**Pull request by ninjanomnom**
Checks if the stationary dock being moved is the same as the old_dock, no more locates.

Adds in all args to all uses of ShuttleMove

Cleans up a couple other procs in the file.

### CODE REVIEW

File: code/modules/shuttle/on_move.dm [hunk 1]
```diff
@@ -1,18 +1,34 @@
...
+#define TURF_ON onShuttleMove(turf/newT, turf_type, baseturf_type, rotation, list/movement_force, move_dir)
+#define TURF_AFTER afterShuttleMove(turf/oldT)
+
+#define ATOM_BEFORE beforeShuttleMove(turf/newT, rotation)
+#define ATOM_ON onShuttleMove(turf/newT, turf/oldT, rotation, list/movement_force, move_dir, old_dock)
+#define ATOM_AFTER afterShuttleMove(list/movement_force, shuttle_dir, shuttle_preferred_direction, move_dir)
+
+#define AREA_BEFORE beforeShuttleMove()
+#define AREA_ON onShuttleMove(turf/oldT, turf/newT, area/underlying_old_area)
+#define AREA_AFTER afterShuttleMove()
```
Comment (ninjanomnom) [inline code/modules/shuttle/on_move.dm]: I feel terrible for using defines like this but it's so damn convenient and it's only for this file.

Comment (MrPerson) [inline code/modules/shuttle/on_move.dm]: You can always just find-and-replace the defines yourself with the real shit anyway. You keep your convenience, we keep our readability, everyone wins

**Review by Cyberboss — DISMISSED** [review]
Yeah, let's not introduce this style here please

File: code/modules/shuttle/on_move.dm (hunk 1 above)
Comment (ninjanomnom) [inline code/modules/shuttle/on_move.dm]: The reasoning behind it here is the procs are named the same for the different types but have different args. But fine, I have an alternative I can try.

**Pull request merged**